src/main/java/de/fearnixx/jeak/teamspeak/voice/connection/VoiceConnection.java

```diff
@@ -0,0 +1,223 @@
+                                        clientTextMessage.setCaption(EventCaptions.TEXT_MESSAGE);
+                                        clientTextMessage.setConnection(bot.getServer().getConnection());
+                                        clientTextMessage.setProperty(PropertyKeys.TextMessage.MESSAGE, e.getMessage());
+                                        clientTextMessage.setProperty("invokerid", invokerId);
+                                        clientTextMessage.setProperty("invokeruid", client.getClientUniqueID());
+                                        clientTextMessage.setProperty("invokername", client.getNickName());
+                                        clientTextMessage.setProperty("targetmode", e.getTargetMode());
+
+                                        eventService.fireEvent(clientTextMessage);
+                                    }
+                                }
+                            }
+                    );
+
+                    InetSocketAddress inetSocketAddress = new InetSocketAddress(hostname, port);
+                    try {
+                        ts3jClientSocket.connect(inetSocketAddress, null, 10000L);
+                    } catch (IOException | TimeoutException e) {
+                        onError.run();
+                        return;
+                    }
+
+                    this.connected = true;
+                    onSuccess.run();
+                }
+        ).start();
+    }
+
+    @Override
+    public void disconnect() {
+        disconnect(null);
+    }
+
+    @Override
+    public void disconnect(String reason) {
+        try {
+            if (ts3jClientSocket.getMicrophone() != null
+                    && ts3jClientSocket.getMicrophone().getClass().isAssignableFrom(IAudioPlayer.class)) {
+                ((IAudioPlayer) ts3jClientSocket.getMicrophone()).stop();
+            }
+
+            ts3jClientSocket.disconnect(reason);
+        } catch (InterruptedException ie) {
+            Thread.currentThread().interrupt();
+        } catch (IOException | ExecutionException | TimeoutException e) {
+            //We assume that the server is not reachable for this connection. Therefor it is declared disconnected
+        }
+
+        locked = false;
+        connected = false;
+    }
+
+    @Override
+    public boolean isConnected() {
+        return connected;
+    }
+
+    @Override
+    public void sendToChannel(int channelId) {
+        sendToChannel(channelId, null);
+    }
+
+    @Override
+    public void sendToChannel(int channelId, String password) {
+        try {
+            ts3jClientSocket.clientMove(ts3jClientSocket.getClientId(), channelId, password);
+        } catch (InterruptedException ie) {
+            Thread.currentThread().interrupt();
+        } catch (IOException | TimeoutException | CommandException e) {
+            //The client movement failed
+
+            //TODO: return the result of an action performed on the connection
```

Comment:
If ``#clientMove`` is blocking, you may return ``false`` from catch blocks. If not, do nothing but logging.

Reply:
Also check the information provided by ``CommandException``.